server/src/main/java/org/folio/shared/index/api/SharedIndexService.java

```diff
@@ -0,0 +1,180 @@
+package org.folio.shared.index.api;
+
+import io.vertx.core.Future;
+import io.vertx.core.Vertx;
+import io.vertx.core.json.JsonArray;
+import io.vertx.core.json.JsonObject;
+import io.vertx.ext.web.Router;
+import io.vertx.ext.web.RoutingContext;
+import io.vertx.ext.web.openapi.RouterBuilder;
+import io.vertx.ext.web.validation.RequestParameter;
+import io.vertx.ext.web.validation.RequestParameters;
+import io.vertx.ext.web.validation.ValidationHandler;
+import java.util.function.Function;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.folio.okapi.common.HttpResponse;
+import org.folio.shared.index.matchkey.MatchKeyMethod;
+import org.folio.shared.index.storage.Storage;
+import org.folio.tlib.RouterCreator;
+import org.folio.tlib.TenantInitHooks;
+import org.folio.tlib.postgres.PgCqlField;
+import org.folio.tlib.postgres.PgCqlQuery;
+
+public class SharedIndexService implements RouterCreator, TenantInitHooks {
+
+  private static final Logger log = LogManager.getLogger(SharedIndexService.class);
+  final Vertx vertx;
+
+  public SharedIndexService(Vertx vertx) {
+    this.vertx = vertx;
+  }
+
+  static String stringOrNull(RequestParameter requestParameter) {
+    return requestParameter == null ? null : requestParameter.getString();
+  }
+
+  Future<Void> putSharedRecords(RoutingContext ctx) {
+    Storage storage = new Storage(ctx);
+    return storage.upsertSharedRecords(ctx.getBodyAsJson()).onSuccess(res -> {
+      JsonArray ar = new JsonArray();
+      // global ids and match keys here ...
+      HttpResponse.responseJson(ctx, 200).end(ar.encode());
+    });
+  }
+
+  Future<Void> getSharedRecords(RoutingContext ctx) {
+    PgCqlQuery pgCqlQuery = PgCqlQuery.query();
+    pgCqlQuery.addField(
+        new PgCqlField("cql.allRecords", PgCqlField.Type.ALWAYS_MATCHES));
+    pgCqlQuery.addField(
+        new PgCqlField("id", PgCqlField.Type.UUID));
+    pgCqlQuery.addField(
+        new PgCqlField("local_identifier", "localId", PgCqlField.Type.TEXT));
+    pgCqlQuery.addField(
+        new PgCqlField("library_id", "sourceId", PgCqlField.Type.UUID));
```

Comment:
Can we align the DB schema with the JSON Schema and params?

Reply:
Oh yes.. We have many mixes of terminology. I'll rename the things in the DB.

Reply:
Then let's use underscored versioned in the DB

Reply:
Done